x-pack/plugins/encrypted_saved_objects/server/create_migration.ts

```diff
@@ -80,20 +92,40 @@ export const getCreateMigration = (
     const encryptDescriptor = { id, type, namespace: encryptedDoc.namespace };
 
     // decrypt the attributes using the input type definition
-    // then migrate the document
-    // then encrypt the attributes using the migration type definition
-    return mapAttributes(
-      migration(
-        mapAttributes(encryptedDoc, (inputAttributes) =>
-          inputService.decryptAttributesSync<any>(decryptDescriptor, inputAttributes, {
+    // if an error occurs during decryption, use the shouldMigrateIfDecryptionFails flag
+    // to determine whether to throw the error or continue the migration
+    // if we are continuing the migration, strip encrypted attributes from the document using stripOrDecryptAttributesSync
+    const documentToMigrate = mapAttributes(encryptedDoc, (inputAttributes) => {
+      try {
+        const decryptedAttributes = inputService.decryptAttributesSync<any>(
+          decryptDescriptor,
+          inputAttributes,
+          { convertToMultiNamespaceType }
+        );
+        return decryptedAttributes;
+      } catch (err) {
+        if (!shouldMigrateIfDecryptionFails || !(err instanceof EncryptionError)) {
+          throw err;
+        }
+
+        context.log.warn(
+          `Decryption failed for encrypted Saved Object "${encryptedDoc.id}" of type "${encryptedDoc.type}" with error: ${err.message}. Migration will be applied to the original encrypted document but this may cause decryption errors later on.`
```

Comment:
__question:__ `Migration will be applied to the original encrypted document but this may cause decryption errors later on.` - maybe we should say here that all encrypted attributes were stripped instead or something along these lines?

Reply:
Updated in https://github.com/elastic/kibana/pull/105968/commits/1365d05c60aeae8c4bd64962cfa6f93293dc4e0d.
Was trying to make it clear without making it too long, WDYT?

Reply:
> but this may cause decryption errors later on

Technically nothing should fail decryption anymore as we stripped all attributes we couldn't decrypt. That made me wonder why in integration tests you still receive 500?

Reply:
And I know why: in tests we return `500` if we cannot find SO (weird decision, but not relevant), and `/api/saved_objects/get-decrypted-as-internal-user/alert` cannot find alert because it's hidden SO! I guess we should use `/api/hidden_saved_objects/get-decrypted-as-internal-user/alert` instead (and likely same thing for `action`). 

Would you mind updating the API integration test (and this message as well, since decryption is not supposed to fail after stripping)?

Reply:
Good catch about the test! Updated the warning text slightly and fixed the test in https://github.com/elastic/kibana/pull/105968/commits/7c55cf95afd841a0fb2e6d798002c7a42dbcaab7